.gitlab-ci.yml

```diff
@@ -104,7 +101,7 @@ test:debug:
   script:
     - rm -rf ~/.gradle/daemon/
     - export DD_AGENT_HOST="$BUILDENV_HOST_IP"
-    - GRADLE_OPTS="-Xmx3072m" ./gradlew :unitTestDebug --stacktrace --no-daemon --build-cache --gradle-user-home cache/
```

Comment:
I added `DD_ENV`, `DD_TAGS` and other config params as agent properties rather than env vars because previously they were only set for the test tasks. The idea was to err on the side of caution, since I was not sure what the implications of setting them for the entire pipeline will be. 

If there are no other jobs that need their own `DD_ENV` and `DD_TAGS`, then I'd rather move these settings to the `variables` section.